### PR TITLE
Option --emergency-codes

### PR DESCRIPTION
**Pull request by Lo0k**
Hello,

This adds the ability to choose the quantity of emergency codes to generate (#29), when running google-authenticator.

Usage: --emergency-codes=N or -e N.
The value of can be less or equal to 10 (hardcoded in MAX_SCRATCHCODES variable).

Tony

### CODE REVIEW

File: man/google-authenticator.1.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+google-authenticator [*options*]
+
+If no option is provided on the command line, google-authenticator(1) will ask interactively the user for the more important options.
+
+# DESCRIPTION
+
+The google-authenticator(1) command creates a new secret key in the current user's home directory. By default, this secret key and all settings will be stored in `~/.google_authenticator`.
+
+If the system supports the `libqrencode` library, a QRCode will be shown, that can be scanned using the Android Google Authenticator application. If the system does not have this library, google-authenticator(1) outputs an URL that can be followed using a web browser. Alternatively, the alphanumeric secret key is also outputed and thus can be manually entered into the Android Google Authenticator application.
```
Comment (ThomasHabets) [inline man/google-authenticator.1.md]: break lines on before 80 char boundary.

File: src/google-authenticator.c [hunk 2]
```diff
@@ -674,6 +691,9 @@ int main(int argc, char *argv[]) {
     fprintf(stderr, "Must set -r when setting -R, and vice versa\n");
     _exit(1);
   }
+  if (emergency_codes == 0) {
+    emergency_codes = MAX_SCRATCHCODES;
```
Comment (ThomasHabets) [inline src/google-authenticator.c]: continue to default to 5

**Comment by ThomasHabets** [comment]
Please also either add manpage generation, or give me cmdline for how to generate it and I'll add it. Also some Makefile.am to *install* the manpage.

Since we'll check in the generated manpage though, it should not be built by mere "make". Maybe "make doc".

**Comment by Lo0k** [comment]
The cmdline requires a dep to pandoc:
pandoc --standalone --to man google-authenticator.1.md > google-authenticator.1

**Comment by ThomasHabets** [comment]
I pulled down the changes, cleaned up, and added 'make doc' and installing the manpage.

Everything from this pull request should be in there. Thank you.

**Pull request closed without merging**